compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1166,11 +1168,16 @@ public void initPredeclaredModules(Map<Name, BPackageSymbol> predeclaredModules,
 
     @Override
     public void visit(BLangXMLNS xmlnsNode) {
+        defineXMLNS(env, xmlnsNode);
+    }
+
+    public void defineXMLNS(SymbolEnv symEnv, BLangXMLNS xmlnsNode) {
         String nsURI = "";
         if (xmlnsNode.namespaceURI.getKind() == NodeKind.SIMPLE_VARIABLE_REF) {
             BLangSimpleVarRef varRef = (BLangSimpleVarRef) xmlnsNode.namespaceURI;
             if (Symbols.isFlagOn(varRef.symbol.flags, Flags.CONSTANT)) {
-                nsURI = ((BConstantSymbol) varRef.symbol).value.toString();
+                BLangConstantValue constantValue = ((BConstantSymbol) varRef.symbol).value;
+                nsURI = constantValue != null ? constantValue.toString() : "";
```

Comment:
When can this be null now?

Reply:
For cases such as this,
```
const X = G;
xmlns X as ns;
```.
Where G is not defined

Reply:
I believe that setting the value to an empty string is not the correct approach. Doing so results in the following errors for above example you mentioned.
```
ERROR [test.bal:(8:11,8:12)] expression is not a constant expression
ERROR [test.bal:(8:11,8:12)] undefined symbol 'G'
ERROR [test.bal:(9:1,9:15)] cannot bind prefix 'ns' to the empty namespace name

```
These errors suggest that the symbol 'G' is undefined. However, the error `cannot bind prefix ns to the empty namespace name` seems to imply that 'G' is defined but as an empty string. Therefore, in my opinion, the only appropriate error message for this scenario should be:
```
ERROR [test.bal:(8:11,8:12)] undefined symbol 'G'
```

Reply:
+1 for skipping the error message `cannot bind prefix 'ns' to the empty namespace name`. What about the error message `expression is not a constant expression`. We should emit that right? Its for the line `const X  = G` where G is also not a constant expression.